### PR TITLE
Add debugging to Arduino code

### PR DESCRIPTION
**Pull request by WestfW**
While TinyUSB in general has very good logging capabilities, the Arduino-specific code will silently fail under a number of circumstances.  This PR adds some logging to at least a couple of those.

It also makes two "logic" changes:
1) Define CFG_TUD_EP_MAX as 8 in src/arduino/ports/samd/tusb_config_samd.h
2) Take a failure return slightly earlier in Adafruit_USBD_Device::addInterface() if the number of endpoints exceeds the max.

### CODE REVIEW

**Comment by WestfW** [comment]
Huh.  CFG_TUD_EP_MAX doesn't have a default definition in any of the typical .h files.  Instead, it checks buried down in device/usbd.c, so it's not currently defined for the non-SAMD platforms. :-(  (Causing the build failures)
```
../device/usbd.c:40:#ifndef CFG_TUD_EP_MAX
../device/usbd.c:41:#define CFG_TUD_EP_MAX          9
```

Shouldn't the default value be defined in tusb_option.h with the rest of the CFG parameters?

File: src/arduino/Adafruit_USBD_Device.cpp [hunk 1]
```diff
@@ -165,11 +166,17 @@ bool Adafruit_USBD_Device::addInterface(Adafruit_USBD_Interface &itf) {
       }
     } else if (desc[1] == TUSB_DESC_ENDPOINT) {
       tusb_desc_endpoint_t *desc_ep = (tusb_desc_endpoint_t *)desc;
+      if (_epin_count >= CFG_TUD_EP_MAX || _epout_count >= CFG_TUD_EP_MAX) {
+        TU_LOG1("%s: Failed: Max endpoints (%d) exceeded\r\n",
+                __PRETTY_FUNCTION__, CFG_TUD_EP_MAX);
+        return false;
+      }
```
Comment (hathach) [inline src/arduino/Adafruit_USBD_Device.cpp:173]: this error log will be printed by tinyusb stack later on, I am working on tthat patch just now. Would you mind removing these.

File: src/arduino/ports/samd/tusb_config_samd.h [hunk 2]
```diff
@@ -85,6 +85,9 @@ extern "C" {
 #define CFG_TUD_VENDOR_RX_BUFSIZE 64
 #define CFG_TUD_VENDOR_TX_BUFSIZE 64
 
+// Vendor Max number of (bidirectional) endpoints
+#define CFG_TUD_EP_MAX 8
```
Comment (hathach) [inline src/arduino/ports/samd/tusb_config_samd.h:89]: same as above, there is a upcoming patch to address these endpoint number. Please revert it.

**Review by hathach — CHANGES_REQUESTED** [review]
Thank you very much for the PR, there is an on-going patch to tinyusb to better log with off limit endpoint number. These check should be dropped here to prevent duplication.

**Review by hathach — APPROVED** [review]
superb !! Look good now. 

FYI, the work in progress is merged in tinyusb stack https://github.com/hathach/tinyusb/pull/975

PS: look like you have issue with clang. please run `clang-format -i src/arduino/*` and commit changes to fix it.

**Comment by hathach** [comment]
Closed due to lack of response from user. And I can’t push to the fork

**Pull request closed without merging**